drivers/sensor/ti/ina23x/ina230.c

```diff
@@ -168,13 +201,26 @@ static int ina230_attr_get(const struct device *dev, enum sensor_channel chan,
 	return 0;
 }
 
+static bool ina230_is_high_precision(const struct ina230_config *config)
+{
+	const struct ina23x_config *priv_data = config->priv_data;
+
+	return priv_data->have_high_precision && config->high_precision;
+}
+
 static int ina230_calibrate(const struct device *dev)
 {
 	const struct ina230_config *config = dev->config;
+	uint16_t cal = config->cal;
 	int ret;
 
+	/* divide by 4 as stated in DS if high-precision mode is enabled */
+	if (ina230_is_high_precision(config)) {
+		cal >>= 2;
+	}
+
 	/* See datasheet "Programming" section */
-	ret = ina23x_reg_write(&config->bus, INA230_REG_CALIB, config->cal);
+	ret = ina23x_reg_write(&config->bus, INA230_REG_CALIB, cal);
```

Comment:
This seems like an unnecessary change for this commit. I think it should either be reverted or have it's own commit with a change reason.

Reply:
The reason is the high-precision option that can be enabled for INA236 and it requires to divide the `SHUNT_CAL` by 4. Do you have a better idea of how to do this?

Reply:
I mean adding a variable to hold the value of `config->cal`. There's no functional change here, just coding style.

Reply:
I think I do not understand you, but I reverted the `cal` usage and used `config->cal` instead. The division of the LSB is done when creating an instance of the device.